example_queries/breadth_histogram_filter.cc

```diff
@@ -230,7 +230,7 @@ void BidiContext::onResponseHeadersInbound() {
     LOG_WARN("x-wasm-property: " + properties_joined);
   }
 
-  if (root_->getWorkloadName() == "productpagev1") {
+  if (root_->getWorkloadName() == ""productpage-v1"") {
```

Comment:
Why are there two sets of double quotes here?

Reply:
Yeah this is a tricky problem, which I have not been able to properly solve yet. Essentially, for a query like: 

`MATCH n-->m: e, WHERE a.service_name == "productpage-v1", RETURN count,`

You would like to get the text between the quotes as identifier. Ideally, the strings can be removed with a `replace` call after the token has been classified as identifier. Simple and fairly efficient. However,  I have not been able to get this past the borrow checker yet. Probably, for good reason, since `replace` allocates a new string. 

The second option is to prune the strings directly using Regex, but this has also problems. First, Rust does not seem to support lookahead Regex, so you are always forced to use two capture groups and choose the capture that does not have the strings with it. Second, even if you have the version without strings you still need to classify it as an identifier. There you need to distinguish between tokens, keywords, and actual identifiers to avoid conflict. One possibility is to augment the `get_single_token` function with information whether the passed `tok_str` used to have string wrappers or not.

I prefer option 1 since it is cleaner but it does not seem possible. Will try to get this done before the meeting.

Reply:
OK, thanks for the explanation; that helps.

1. Stepping back a few levels up, do we really need to support hyphens in identifier names? Not many languages support that. 

2. If we do decide to support hyphens in identifier names, let's go with the current double double-quotes strategy. The double double quotes at least makes it clear that the string within the inner double quotes is a literal (and not say productpage minus v1). If we use the double double-quotes strategy, do we still need the replace/regex options you described above?

Reply:
Regarding point 1 hyphens seem to be fairly common in Kubernetes names and are explicitly allowed. The pattern for names is `[a-z0-9]([-a-z0-9]*[a-z0-9])?` according to [this](https://stackoverflow.com/a/51724905) code. 

I pushed an approach that works reasonable well for now.